staging/src/k8s.io/apiserver/pkg/storage/value/transformer.go

```diff
@@ -49,6 +49,10 @@ type identityTransformer struct{}
 var IdentityTransformer Transformer = identityTransformer{}
 
 func (identityTransformer) TransformFromStorage(b []byte, ctx Context) ([]byte, bool, error) {
+	// Other providers do not have k8s prefix
+	if len(b) > 0 && b[0] != '{' && bytes.HasPrefix(b, []byte("k8s:enc:")) {
```

Comment:
This is no longer really the identity transformer (truly generic), it's one that is specific to encryption because it is checking against the encryption prefix.  It would be better to have this in a separate package dedicated to that, and add a better comment about the known types.  It's somewhat of an encapsulation violation because this method now has to know about all possible encodings, but I can live with that for a while.